PMC Tools: Couldn't load assembly 'System.ComponentModel.Annotations'

Running PMC commands with the 2.1 nightlies has a binding redirect error:

Forcing a lower System.ComponentModel.Annotations and ignoring the nuget warnings makes it work.

```
PM> update-database
System.IO.FileLoadException: Could not load file or assembly 'System.ComponentModel.Annotations, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'System.ComponentModel.Annotations, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.CoreConventionSetBuilder.CreateConventionSet()
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.CreateConventionSet(IConventionSetBuilder conventionSetBuilder)
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.CreateModel(DbContext context, IConventionSetBuilder conventionSetBuilder, IModelValidator validator)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.CreateModel()
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.get_Model()
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass1_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
   at Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
   at Microsoft.EntityFrameworkCore.Infrastructure.AccessorExtensions.GetService[TService](IInfrastructure`1 accessor)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(Func`1 factory)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(String contextType)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.UpdateDatabase(String targetMigration, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

Could not load file or assembly 'System.ComponentModel.Annotations, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
PM> 
```

### Steps to reproduce
Include a complete code listing (or project/solution) that we can run to reproduce the issue.

Partial code listings, or multiple fragments of code, will slow down our response or cause us to push the issue back to you to provide code to reproduce the issue.

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net461</TargetFramework>
  </PropertyGroup>
  
  <ItemGroup>
    <!-- uncommenting this, makes it work -->
    <!--<PackageReference Include="System.ComponentModel.Annotations" Version="4.4.0">
      <NoWarn>NU1605</NoWarn>
    </PackageReference>-->
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.1.0-preview1-27959">
      <NoWarn>NU1605</NoWarn>
    </PackageReference>
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="2.1.0-preview1-27959" PrivateAssets="true">
      <NoWarn>NU1605</NoWarn>
    </PackageReference>
  </ItemGroup>

</Project>
```

### Further technical details
EF Core version: 2.1 nightlies
Database Provider: (e.g. Microsoft.EntityFrameworkCore.SqlServer)
Operating system: 
IDE: Visual Studio 2017 15.6 preview1

@bricelam 
```
"myproject.csproj" (Restore target) (1) ->
(Restore target) ->
  myproject.csproj : error NU1605: Detected package downgrade: System.ComponentModel.Annotations from 4.5.0-preview1-26016-05 to 4.4.0. Reference the package directly from the project to select a different version.
myproject.csproj : error NU1605:  myproject -> Microsoft.EntityFrameworkCore.SqlServer 2.1.0-preview1-27959 -> Microsoft.EntityFrameworkCore.Relational 2.1.0-preview1-27959 -> Microsoft.EntityFrameworkCore 2.1.0-preview1-27959 -> System.ComponentModel.Annotations (>= 4.5.0-preview1-26016-05)
myproject.csproj : error NU1605:  myproject -> System.ComponentModel.Annotations (>= 4.4.0)
  myproject.csproj : error NU1605: Detected package downgrade: System.ComponentModel.Annotations from 4.5.0-preview1-26016-05 to 4.4.0. Reference the package directly from the project to select a different version.
myproject.csproj : error NU1605:  myproject -> Microsoft.EntityFrameworkCore.Tools 2.1.0-preview1-27959 -> Microsoft.EntityFrameworkCore.Design 2.1.0-preview1-27959 -> Microsoft.EntityFrameworkCore.Relational 2.1.0-preview1-27959 -> Microsoft.EntityFrameworkCore 2.1.0-preview1-27959 -> System.ComponentModel.Annotations (>= 4.5.0-preview1-26016-05)
myproject.csproj : error NU1605:  myproject -> System.ComponentModel.Annotations (>= 4.4.0)
```

--- a/NuGetPackageVerifier.json
+++ b/NuGetPackageVerifier.json
@@ -4,6 +4,13 @@
             "AdxVerificationCompositeRule"
         ],
         "packages": {
+            "Microsoft.EntityFrameworkCore.Design": {
+                "Exclusions": {
+                    "BUILD_ITEMS_FRAMEWORK": {
+                        "*": "False negative. aspnet/BuildTools#554"
+                    }
+                }
+            },
             "Microsoft.EntityFrameworkCore.Tools.DotNet": {
                 "packageTypes": [
                     "DotnetCliTool"

--- a/src/EFCore.Design/EFCore.Design.csproj
+++ b/src/EFCore.Design/EFCore.Design.csproj
@@ -14,6 +14,13 @@
   </ItemGroup>
 
   <ItemGroup>
+    <None Include="build\**\*">
+      <Pack>True</Pack>
+      <PackagePath>build</PackagePath>
+    </None>
+  </ItemGroup>
+  
+  <ItemGroup>
     <ProjectReference Include="..\EFCore.Relational\EFCore.Relational.csproj" />
   </ItemGroup>
 

--- a/src/EFCore.Design/build/net461/Microsoft.EntityFrameworkCore.Design.props
+++ b/src/EFCore.Design/build/net461/Microsoft.EntityFrameworkCore.Design.props
@@ -1,0 +1,6 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <AutoGenerateBindingRedirects>True</AutoGenerateBindingRedirects>
+    <GenerateBindingRedirectsOutputType>True</GenerateBindingRedirectsOutputType>
+  </PropertyGroup>
+</Project>

--- a/src/EFCore.Design/build/netcoreapp2.0/Microsoft.EntityFrameworkCore.Design.props
+++ b/src/EFCore.Design/build/netcoreapp2.0/Microsoft.EntityFrameworkCore.Design.props
@@ -1,0 +1,5 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <GenerateRuntimeConfigurationFiles>True</GenerateRuntimeConfigurationFiles>
+  </PropertyGroup>
+</Project>

--- a/src/EFCore.Tools/EFCore.Tools.nuspec
+++ b/src/EFCore.Tools/EFCore.Tools.nuspec
@@ -13,7 +13,7 @@
     <repository type="git" url="https://github.com/aspnet/EntityFramework.Tools.git" />
     <dependencies>
       <group targetFramework=".NETStandard2.0">
-        <dependency id="Microsoft.EntityFrameworkCore.Design" version="$version$" exclude="Build,Analyzers" />
+        <dependency id="Microsoft.EntityFrameworkCore.Design" version="$version$" exclude="Analyzers" />
       </group>
     </dependencies>
     <licenseUrl>https://raw.githubusercontent.com/aspnet/Home/2.0.0/LICENSE.txt</licenseUrl>

--- a/src/EFCore.Tools/tools/EntityFrameworkCore.psm1
+++ b/src/EFCore.Tools/tools/EntityFrameworkCore.psm1
@@ -761,8 +761,7 @@ function EF($project, $startupProject, $params, [switch] $skipBuild)
     }
 
     $startupProjectDir = GetProperty $startupProject.Properties 'FullPath'
-    $activeConfiguration = $startupProject.ConfigurationManager.ActiveConfiguration
-    $outputPath = GetProperty $activeConfiguration.Properties 'OutputPath'
+    $outputPath = GetProperty $startupProject.ConfigurationManager.ActiveConfiguration.Properties 'OutputPath'
     $targetDir = Join-Path $startupProjectDir $outputPath -Resolve | Convert-Path
     $startupTargetFileName = GetOutputFileName $startupProject
     $startupTargetPath = Join-Path $targetDir $startupTargetFileName
@@ -808,15 +807,6 @@ function EF($project, $startupProject, $params, [switch] $skipBuild)
             $projectAssets.packageFolders.psobject.Properties.Name | %{
                 $dotnetParams += '--additionalprobingpath', $_.TrimEnd('\')
             }
-        }
-
-        if (!(Test-Path $runtimeConfig))
-        {
-            $configuration = $activeConfiguration.ConfigurationName
-            $platformTarget = GetPlatformTarget $startupProject
-            $projectPath = $startupProject.FullName
-
-            dotnet msbuild /t:GenerateBuildRuntimeConfigurationFiles "/p:GenerateRuntimeConfigurationFiles=True;Configuration=$configuration;Platform=$platformTarget" /verbosity:quiet /nologo $projectPath | Out-Null
         }
 
         if (Test-Path $runtimeConfig)

--- a/src/dotnet-ef/Project.cs
+++ b/src/dotnet-ef/Project.cs
@@ -167,7 +167,6 @@ namespace Microsoft.EntityFrameworkCore.Tools
                 args.Add(_runtime);
             }
 
-            args.Add("/p:GenerateRuntimeConfigurationFiles=True");
             args.Add("/verbosity:quiet");
             args.Add("/nologo");
 

